packages/webpack-stats-differ/src/getFriendlyAssetName.ts

```diff
@@ -37,38 +36,12 @@ export function getFriendlyAssetName(
   return name;
 }
 
-const hashRegex = /_([a-z0-9]{20})/;
+const removeHashFromName = (name: string): string => {
+  return name.replace(/([_.][a-z0-9]{20})(?:\b)/, "");
+};
 
-// TODO(mapol): We need a better way to identify a hash inside a filename
-// TODO(mapol): We need to make this function take into account the folder name. Once we do that we can remove the `ignoreDelveStringsFiles` function.
-export const removeHashFromName = (name: string): string => {
+const cleanFileName = (name: string): string => {
   const fileParts = upath.parse(name);
-  // We could have filenames like .d.ts or .js.map
-  const splitBySecondDotPart = fileParts.name.split(".");
-  const firstNamePart = splitBySecondDotPart[0];
-  const secondDotPart = splitBySecondDotPart[1];
-  // Remove hash
-  const otherParts = firstNamePart.split("_");
-
-  if (
-    otherParts.length === 1 ||
-    otherParts[otherParts.length - 1].length !== 20
-  ) {
-    // Search uses a hash in the secondDotPart e.g. search_init.min_7ba0dc2ea2246e82e8a5.js
-
-    return (
-      [
-        otherParts.join("_"),
-        secondDotPart && secondDotPart.replace(hashRegex, ""),
-      ]
-        .filter(identity)
-        .join(".") + fileParts.ext
-    );
-  }
-
-  return (
-    [otherParts.slice(0, -1).join("_"), secondDotPart]
-      .filter(identity)
-      .join(".") + fileParts.ext
-  );
-};
+  const baseName = fileParts.name;
+  return removeHashFromName(baseName) + fileParts.ext;
```

Comment:
If we only use upath for this, then let's get rid of it and update the string replace.
```typescript
export const removeHashFromName = (name: string): string => {
 return name.replace(/^.*[\/]|([_.][a-z0-9]{20})(?:\b)/g, ""));
};
```

Reply:
Otherwise, sure, let's go with this.